Client/Frontend/Fakespot/Views/FakespotAdView.swift

```diff
@@ -164,8 +206,15 @@ class FakespotAdView: UIView, Notifiable, ThemeApplicable, UITextViewDelegate {
         )
         productLinkButton.configure(viewModel: productLinkButtonViewModel)
         gradeReliabilityScoreView.configure(grade: productAdsData.grade)
+        defaultImageView.accessibilityIdentifier = viewModel.defaultImageA11yId
+        productImageView.accessibilityIdentifier = viewModel.productImageA11yId
+        productImageView.isHidden = true
+
         Task {
-            try? await productImageView.loadImage(from: productAdsData.imageUrl)
+            try? await viewModel.loadImage(from: productAdsData.imageUrl) { [weak self] image in
+                 self?.productImageView.image = image
```

Comment:
async and completion block based code are opposite implementations so I would expect either await for the image or we get it through completion.